drivers/dma/dma_stm32f4x.c

```diff
@@ -523,6 +523,21 @@ static int dma_stm32_stop(struct device *dev, u32_t id)
 	return 0;
 }
 
+static int dma_stm32_get_status(struct device *dev, u32_t id,
+				struct dma_status *stat)
+{
+	struct dma_stm32_device *ddata = dev->driver_data;
+
+	if (id >= DMA_STM32_MAX_STREAMS && stat != NULL) {
```

Comment:
The check looks wrong. I think you meant
````
if (id >= DMA_STM32_MAX_STREAMS || stat == NULL) {
````

Reply:
Thanks for capturing!